bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/custom/CommonWidgetsDPIChangeHandlers.java

```diff
@@ -43,4 +46,24 @@ private static void handleItemDPIChange(Widget widget, int newZoom, float scalin
 			item.setImage(image);
 		}
 	}
+
+	private static void handleStyledTextDPIChange(Widget widget, int newZoom, float scalingFactor) {
+		if (!(widget instanceof StyledText)) {
+			return;
+		}
+		StyledText styledText = (StyledText) widget;
+
+		DPIZoomChangeRegistry.applyChange(styledText.getCaret(), newZoom, scalingFactor);
+		styledText.getCaret().setSize(SWT.DEFAULT, styledText.getLineHeight());
```

Comment:
Isn't this superfluous? I mean the `for`-loop below probably does it too, right?

Reply:
Ok